### ISSUE TITLE
Rule proposal: make padding-line-between-statements support TS syntax

### ISSUE THREAD

**Issue by pascalgn** [typescript-eslint/typescript-eslint#2293]
**Proposal**

Currently, [padding-line-between-statements](https://eslint.org/docs/rules/padding-line-between-statements) supports generic JS syntax elements like "class", "block", "block-like", "switch", "throw". However, TS-specific elements like "type" or "interface" are not supported.

```JSON
{
  "rules": {
    "@typescript-eslint/padding-line-between-statements": [
      "error",
      {
        "blankLine": "always",
        "prev": "type",
        "next": "interface"
      }
    ]
  }
}
```

```TS
type SomeType = {
  value: string;
};
interface SomeInterface {
}
```

**Expected Result**

Rule should report the missing blank line between _SomeType_ and _SomeInterface_.

**Actual Result**

The padding-line-between-statements rule from ESLint [doesn't support](https://github.com/eslint/eslint/blob/master/lib/rules/padding-line-between-statements.js#L350) the configuration options "type" or "interface" and will throw an error.

**Additional Info**

Original issue was submitted to eslint/eslint first: https://github.com/eslint/eslint/issues/13477

**Comment by bradzacher** [comment]
happy to accept a PR!

**Comment by gomesalexandre** [comment]
Will take a look ASAP

**Comment by brentschroeter** [comment]
Is this still fair game for a PR?

**Comment by bradzacher** [comment]
Feel free - Nobody's put one up yet!

**Comment by jun-sheaf** [comment]
I've PRed new adjustments to the original rule which will (coincidentally) allow for TS syntax support. See https://github.com/eslint/eslint/pull/13925

**Comment by jun-sheaf** [comment]
So `eslint` has frozen new options to their stylistic plugins. I could PR this plugin in this repository again, however given the changes that should be made to the original, I've forked the plugin and built a better one: [eslint-plugin-padding](https://github.com/mu-io/eslint-plugin-padding)

Cheers 🎉

**Comment by yasarsid** [comment]
@bradzacher - I don't see even the base rule being part of the typescript-eslint. Can we try and get the base rule out and then look at these improvements ?

**Comment by bradzacher** [comment]
With any issue in opened in this project - it either has a visible progress in the form of an attached PR, or it has no progress.

We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.

If not - please just subscribe to the issue and wait patiently.
Commenting asking for status updates does not bump issue priority in any way and just serves to spam ***everyone*** subscribed to the issue.

**Comment by yasarsid** [comment]
@bradzacher - I'm sorry if it felt otherwise - but my intentions was NOT to spam or neither get a status update. I wanted to have a conversation on this issue.

The reason for my question was if we can release it without the improvements proposed - and if that is the case and if @jun-sheaf  is not actively working on it. I could pick up address this issue.

This issue is important to us and we think this should be available in typescript-eslint.

**Comment by bradzacher** [comment]
As far as I can tell - jun-sheaf chose to publish their changes as a separate plugin - [eslint-plugin-padding](https://github.com/mu-io/eslint-plugin-padding).

If someone is so motivated to make the required changes to add this extension rule to this project- I'm happy to accept a PR.

**Comment by yasarsid** [comment]
Cool - Thanks will raise a PR for the extension rule
